src/app/custom-navigation-mode/custom-navigation-mode.component.ts

```diff
@@ -0,0 +1,21 @@
+import {Component, OnInit} from '@angular/core';
+import {CustomNavigationMode} from './custom-navigation-mode';
+import {WizardComponent, NavigationMode} from 'angular-archwizard';
+
+@Component({
+  selector: 'app-custom-navigation-mode',
+  templateUrl: './custom-navigation-mode.component.html',
+  styleUrls: ['./custom-navigation-mode.component.css']
+})
+export class CustomNavigationModeComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+  getCustomNavigationMode(wizard: WizardComponent): NavigationMode {
```

Comment:
Is there a reason for using a getter method to retrieve the `CustomNavigationMode` instance?
Instead I think it is better to use a field on the class instance level or even a static field.
I mean you don't want to change the navigation mode later on, right?

Reply:
After https://github.com/madoar/angular-archwizard/pull/166, the wizard can take a custom `[navigationMode]`, but this navigationMode can only be a string or a factory, not an instance.

The problem is that in order to construct an instance, you need a `wizardState`, and if you try to pull it yourself from the wizard yourself like this

```
class MyComponent {
  @ViewChild('wizard') wizard: WizardComponent;

  ngOnInit() {
    this.navigationMode = new CustomNavigationMode(this.wizard.wizardState);
  }
}
```

it may not be initialized yet.

So the wizard takes a factory and calls it to retrieve an instance when the wizard has already initialized `wizardState`.

Everything could be simplified if `NavigationMode` classes did not store `wizardState` inside themselves, but received it as an argument in each method. This is my proposal.

In this case, `NavigationMode`'s will be stateless (at least from the wizard's point of view), and an instance could be passed instead of a factory as you suggested.

Does it make sense? What do you think?

Reply:
Yes the accessibility of the `WizardState` is indeed a problem. The issue is that it is not created yet in the initialization phase. The earliest you can access the `WizardState` is currently during `ngAfterViewInit()`, if I remember correctly.

My favorite solution would be to allow for the injection of the `WizardState`, but I think this is not possible. Therefore I think we should go with your suggested solution, i.e. don't persist a reference to the `WizardState` in the `NavigationMode`.

Reply:
PR for stateless nav mode classes:
https://github.com/madoar/angular-archwizard/pull/203